packages/opentelemetry-test-utils/src/test-utils.ts

```diff
@@ -158,6 +159,27 @@ export const getPackageVersion = (packageName: string) => {
       require?.main?.paths || []
     ),
   });
-  const packageJsonPath = path.join(path.dirname(packagePath), 'package.json');
+
+  // Some packages are resolved to a subfolder because their "main" points to it.
+  // As a consequence the "package.json" path is wrong and we get a MODULE_NOT_FOUND
+  // error. We should walk up the folder structure to lookup for the file.
+  // `tedious` package is an example
+  // {
+  //   "name: "tedious",
+  //   "main: "lib/tedious.js",
+  //   ...
+  // }
+  // resolving `packagePath` to `/path/to/opentelemetry-js-contrib/node_modules/tedious/lib/tedious.js`
+  let packageJsonDir = path.dirname(packagePath);
+  let packageJsonPath = path.join(path.dirname(packagePath), 'package.json');
+
+  while (
+    !existsSync(packageJsonPath) &&
+    !packageJsonDir.endsWith('node_modules') &&
+    packageJsonDir !== '/'
```

Comment:
issue: I guess the check `packageJsonDir !== '/'` is not reliable for all operation systems. You could mabye use `packageJsonDir !== path.resolve(packageJsonDir, '..')` or similar. 

I was wondering why do we need to check at all? 🤔  IMO `!packageJsonDir.endsWith('node_modules')` is enough for the current use case of the function.

Reply:
>  IMO !packageJsonDir.endsWith('node_modules') is enough for the current use case of the function.

I think you're right because `require.resolve` would fail if the string provided does not resolve to a package. So I?m removing the lats check. Thanks!

Reply:
An alternative might be this:

```diff
diff --git a/packages/opentelemetry-test-utils/src/test-utils.ts b/packages/opentelemetry-test-utils/src/test-utils.ts
index aae38f16..7c7914d9 100644
--- a/packages/opentelemetry-test-utils/src/test-utils.ts
+++ b/packages/opentelemetry-test-utils/src/test-utils.ts
@@ -28,6 +28,7 @@ import {
   hrTimeToMilliseconds,
   hrTimeToMicroseconds,
 } from '@opentelemetry/core';
+import * as fs from 'fs';
 import * as path from 'path';

 const dockerRunCmds = {
@@ -153,11 +154,12 @@ export const getPackageVersion = (packageName: string) => {
   // "test-all-versions" tests that tend to install conflicting package
   // versions. Prefix the search paths with the cwd to include the workspace
   // dir.
-  const packagePath = require?.resolve(packageName, {
+  const mainPath = require?.resolve(packageName, {
     paths: [path.join(process.cwd(), 'node_modules')].concat(
       require?.main?.paths || []
     ),
   });
-  const packageJsonPath = path.join(path.dirname(packagePath), 'package.json');
-  return require(packageJsonPath).version;
+  const idx = mainPath.lastIndexOf('node_modules');
+  const pjPath = path.join(mainPath.slice(0, idx), 'node_modules', packageName, 'package.json');
+  return JSON.parse(fs.readFileSync(pjPath, 'utf8')).version;
 };
```

1. For the `pjPath` it:
    - takes resolved "main" path, e.g. `/Users/trentm/tm/opentelemetry-js-contrib4/plugins/node/instrumentation-tedious/node_modules/tedious/lib/tedious.js`, 
    - slices off from the last `node_modules`, e.g.: `/Users/trentm/tm/opentelemetry-js-contrib4/plugins/node/instrumentation-tedious`
    - then tacks on the `node_modules/$packageName/package.json`.
2. I've also changed from `require(pjPath)` to using JSON.parse(fs.read). It isn't *needed* here, but it avoids the rare case where a package defines "exports" and those exports don't allow `require`ing its package.json.

The technique for "1." is basically what the module-details-from-path mini-module is doing. Both RITM and IITM use that module to do basically this.

Reply:
thanks @trent, I think you're right in the assumption that a resolved package must have a `node_modules` and getting the last index short circuits the lookup. Also I did not think about 2 and it's a good addition IMO.

Applying this suggestion in next commit :)